### DISCUSSION TITLE
[Question] Is it possible to return progress info from openvdb?

### DISCUSSION THREAD

**Discussion by CapLek** [AcademySoftwareFoundation/openvdb#1143] [Q&A]
Is it possible to return information about progress from openvdb long running tasks such as grid resampling, offsetting etc.?

**Comment by Idclip** [comment]
Hey @CapLek,

Good question, short answer is no - there is some support for runtime logging if you build with the Log4cplus dependency, but it's pretty limited and we don't tend to use it much. It would be nice if it was used more consistently or if we incorporated something more lightweight that can be used in these long running threaded ops (like spdlog).

**Reply by diiigle** [reply]
The [interface](https://github.com/AcademySoftwareFoundation/openvdb/blob/master/openvdb/openvdb/util/NullInterrupter.h) for it actually exists and it is used in (AFAIK at least) MeshToVolume.h .

https://github.com/AcademySoftwareFoundation/openvdb/blob/ea12eb6c1ee27f78746e01e037950e176ee9e9e5/openvdb/openvdb/tools/MeshToVolume.h#L3232-L3234

**Comment by Idclip** [comment]
Going to migrate this to github discussions as it would be worth discussing this more.
